lib/ssl/src/tls_dtls_connection.erl

```diff
@@ -341,10 +341,12 @@ certify(internal, #certificate{asn1_certificates = [Peer|_]} = Cert,
                                cert_db_ref = CertDbRef,
                                crl_db = CRLDbInfo},
                handshake_env = #handshake_env{
-                                  stapling_state = #{status := Status} = StaplingState},
+                                  stapling_state = #{status := StaplingStatus} =
+                                      StaplingState},
                connection_env = #connection_env{
                                    negotiated_version = Version},
-               ssl_options = Opts} = State0) when Status =/= staple ->
+               ssl_options = Opts} = State0)
+  when StaplingStatus == not_negotiated; StaplingStatus == received_peer ->
```

Comment:
I do not understand why StaplingStatus could be equal to not_negotiated here? This code is not executed in TLS-1.3 context, to my understanding it should only received_peer.

Reply:
Or probably received_response when all changes are applied!

Reply:
Ah, I think new commit clears it up :)  Old behavior was kind of double negation.